### PR TITLE
Passing credentials and secrets

### PR DESCRIPTION
**Pull request by eliskasl**

### CODE REVIEW

File: planning/planning_index.adoc [hunk 1]
```diff
@@ -246,6 +246,43 @@ directory between host and container using the _-v_ flag in your RUN label.
 === Security and User considerations
 
 ==== Passing credentials and secrets
+
+Storing sensitive data like credentials is a hot topic, especially because Docker does not provide a designated option for storing and passing secret values to containers.
+
+Currently, the most popular and reasonable way to pass credentials and secrets in a container is specifying them as **environment variables** at container runtime. You as a consumer of such an image don't expose any of your sensitive data publicly.
```
Comment (cgwalters) [inline planning/planning_index.adoc]: We should at least mention http://kubernetes.io/docs/user-guide/secrets/

File: planning/planning_index.adoc [hunk 2]
```diff
@@ -246,6 +246,43 @@ directory between host and container using the _-v_ flag in your RUN label.
...
+
+There are other ways how to store secrets and although using environment variables might lead to leaking private data in certain corner cases, it still belongs to the safest workarounds available.
+
+There are a couple of things to keep in mind when operating with secrets:
+
+* For obvious reasons, you should avoid using default passwords - users tend to forget to change the default configuration and in case a known password leaks, it can be easily misused.
+
+* Although squashing removes intermediate layers from the final image, secrets from those layers will still be present in the build cache.
+
+// Do we want to suggest https://www.vaultproject.io/intro/index.html ?
```
Comment (cgwalters) [inline planning/planning_index.adoc]: The Red Hat IdM team also has a container secrets project, I forget the name.

Comment (cgwalters) [inline planning/planning_index.adoc]: https://github.com/latchset/custodia

File: planning/planning_index.adoc [hunk 3]
```diff
@@ -246,6 +247,56 @@ directory between host and container using the _-v_ flag in your RUN label.
...
+
+There are a couple of things to keep in mind when operating with secrets:
+
+* For obvious reasons, you should avoid using default passwords - users tend to forget to change the default configuration and in case a known password leaks, it can be easily misused.
+
+* Although squashing removes intermediate layers from the final image, secrets from those layers will still be present in the build cache.
+
+===== Handling Secrets in Kubernetes
+
+Containers running through Kubernetes can take advantage of the **secret** object type to store sensitive data such as passwords or tokens. To learn how to create and access these, refer to the http://kubernetes.io/docs/user-guide/secrets/[Kubernetes User Guide].
```
Comment (mfojtik) [inline planning/planning_index.adoc]: Secret object (it is a resource name)

File: planning/planning_index.adoc [hunk 4]
```diff
@@ -246,6 +247,56 @@ directory between host and container using the _-v_ flag in your RUN label.
 === Security and User considerations
 
 ==== Passing credentials and secrets
+
+Storing sensitive data like credentials is a hot topic, especially because Docker does not provide a designated option for storing and passing secret values to containers.
+
+Currently, the most popular and a reasonable way to pass credentials and secrets in a container is specifying them as **environment variables** at container runtime. You as a consumer of such an image don't expose any of your sensitive data publicly.
```
Comment (mfojtik) [inline planning/planning_index.adoc]: i would not say "reasonable" but doable if you know what are you doing... another way is to use tmpfs volume where you have the secrets stored (which is what kube is doing btw.)

**Comment by mfojtik** [comment]
LGTM

**Pull request merged**